crates/bevy_reflect/src/serde/de/deserializer.rs

```diff
@@ -14,6 +14,193 @@ use crate::{
+/// #     fn with_asset_type_id(&mut self, (): ()) -> &mut Self { unimplemented!() }
+/// #     fn untyped(&mut self) -> &mut Self { unimplemented!() }
+/// #     fn load_asset(&mut self, (): ()) -> Handle<LoadedUntypedAsset> { unimplemented!() }
+/// # }
+/// #
+/// # struct ReflectHandle;
+/// # impl TypeData for ReflectHandle {
+/// #     fn clone_type_data(&self) -> Box<dyn TypeData> {
+/// #         unimplemented!()
+/// #     }
+/// # }
+/// # impl ReflectHandle {
+/// #     fn asset_type_id(&self) {
+/// #         unimplemented!()
+/// #     }
+/// # }
+/// #
+/// # struct AssetPathVisitor;
+/// # impl<'de> Visitor<'de> for AssetPathVisitor {
+/// #     type Value = ();
+/// #     fn expecting(&self, formatter: &mut core::fmt::Formatter) -> core::fmt::Result { unimplemented!() }
+/// # }
+/// # type AssetError = Box<dyn core::error::Error>;
+/// #[derive(Debug, Clone, Reflect)]
+/// struct MyAsset {
+///     name: String,
+///     mesh: Handle<Mesh>,
+/// }
+///
+/// fn load(
+///     asset_bytes: &[u8],
+///     type_registry: &TypeRegistry,
+///     load_context: &mut LoadContext,
+/// ) -> Result<MyAsset, AssetError> {
+///     struct HandleProcessor<'a> {
+///         load_context: &'a mut LoadContext,
+///     }
+///
+///     impl ReflectDeserializerProcessor for HandleProcessor<'_> {
+///         fn try_deserialize<'de, D>(
+///             &mut self,
+///             registration: &TypeRegistration,
+///             _registry: &TypeRegistry,
+///             deserializer: D,
+///         ) -> Result<Result<Box<dyn PartialReflect>, D>, D::Error>
+///         where
+///             D: Deserializer<'de>,
+///         {
+///             let Some(reflect_handle) = registration.data::<ReflectHandle>() else {
+///                 // we don't want to deserialize this - give the deserializer back
+///                 return Ok(Err(deserializer));
+///             };
+///
+///             let asset_type_id = reflect_handle.asset_type_id();
+///             let asset_path = deserializer.deserialize_str(AssetPathVisitor)?;
+///
+///             let handle: Handle<LoadedUntypedAsset> = self.load_context
+///                 .load()
+///                 .with_asset_type_id(asset_type_id)
+///                 .untyped()
+///                 .load_asset(asset_path);
+///             # let _: Result<_, ()> = {
+///             Ok(Box::new(handle))
+///             # };
+///             # unimplemented!()
+///         }
+///     }
+///
+///     let mut ron_deserializer = ron::Deserializer::from_bytes(asset_bytes)?;
+///     let mut processor = HandleProcessor { load_context };
+///     let reflect_deserializer =
+///         ReflectDeserializer::with_processor(type_registry, &mut processor);
+///     let asset = reflect_deserializer.deserialize(&mut ron_deserializer)?;
+///     # unimplemented!()
+/// }
+/// ```
+///
+/// [`try_deserialize`]: Self::try_deserialize
+pub trait ReflectDeserializerProcessor {
```

Comment:
Nit: For the sake of organization and discoverability, I'd prefer if this was in its own module (even if it is super tied to the deserializers themselves).

Reply:
Agreed, moved to `processor.rs` and pub reexported from `mod.rs`